### PR TITLE
doc: Add initial doc how to expand Comet exceptions

### LINKED ISSUE #104: Include Rust's native stack trace in CometNativeException
### What is the problem the feature request solves?

Currently, the backtrace is only added when rust panics.

However, when the exception is thrown back to JVM side. The error msg sometimes might be not that helpful. It would be ideal to get the backtrace for the rust error as well.

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #104 .

## Rationale for this change
Adding docs how to extend Comet exceptions by leveraging DF backtraces

### CODE REVIEW

**Comment by comphead** [comment]
@snmvaughan cc

File: DEBUGGING.md [hunk 1]
```diff
@@ -30,6 +30,62 @@ LLDB or the LLDB that is bundled with XCode. We will use the LLDB packaged with
...
+datafusion-common = { version = "36.0.0", features = ["backtrace"] }
+datafusion = { default-features = false, version = "36.0.0", features = ["unicode_expressions", "backtrace"] }
+```
+
+Then build the Comet as [described](https://github.com/apache/arrow-datafusion-comet/blob/main/README.md#getting-started)
+
+Start Comet with `RUST_BACKTRACE=1`
+
+```commandline
+RUST_BACKTRACE=1 $SPARK_HOME/spark-shell --jars spark/target/comet-spark-spark3.4_2.12-0.1.0-SNAPSHOT.jar --conf spark.sql.extensions=org.apache.comet.CometSparkSessionExtensions --conf spark.comet.enabled=true --conf spark.comet.exec.enabled=true --conf spark.comet.exec.all.enabled=true
```
Comment (sunchao) [inline DEBUGGING.md:61]: Just curious, if we don't enable `RUST_BACKTRACE=1`, is there any performance penalty from enabling `backtrace`?

[the remark this answers is not in the document]
Comment (comphead) [reply DEBUGGING.md:61]: That is good idea if we decide to keep backtrace enabled always

File: DEBUGGING.md [hunk 2]
```diff
@@ -94,3 +94,62 @@ https://mail.openjdk.org/pipermail/hotspot-dev/2019-September/039429.html
...
+org.apache.comet.CometNativeException: Internal error: MIN/MAX is not expected to receive scalars of incompatible types (Date32("NULL"), Int32(15901)).
+This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+        at org.apache.comet.Native.executePlan(Native Method)
+        at org.apache.comet.CometExecIterator.executeNative(CometExecIterator.scala:65)
+        at org.apache.comet.CometExecIterator.getNextBatch(CometExecIterator.scala:111)
+        at org.apache.comet.CometExecIterator.hasNext(CometExecIterator.scala:126)
+
+```
+
+There is a verbose exception option by leveraging Datafusion [backtraces](https://arrow.apache.org/datafusion/user-guide/example-usage.html#enable-backtraces)
```
Comment (viirya) [inline DEBUGGING.md:114]:
```suggestion
There is a verbose exception option by leveraging DataFusion [backtraces](https://arrow.apache.org/datafusion/user-guide/example-usage.html#enable-backtraces)
```

File: DEBUGGING.md [hunk 3]
```diff
@@ -94,3 +94,62 @@ https://mail.openjdk.org/pipermail/hotspot-dev/2019-September/039429.html
...
+This option allows to append native Datafusion stacktrace to the original error message. 
+To enable this option with Comet it is needed to include `backtrace` feature in [Cargo.toml](https://github.com/apache/arrow-datafusion-comet/blob/main/core/Cargo.toml) for Datafusion dependencies
```
Comment (viirya) [inline DEBUGGING.md:116]:
```suggestion
To enable this option with Comet it is needed to include `backtrace` feature in [Cargo.toml](https://github.com/apache/arrow-datafusion-comet/blob/main/core/Cargo.toml) for DataFusion dependencies
```

File: DEBUGGING.md [hunk 4]
```diff
@@ -94,3 +94,62 @@ https://mail.openjdk.org/pipermail/hotspot-dev/2019-September/039429.html
...
+This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+        at org.apache.comet.Native.executePlan(Native Method)
+at org.apache.comet.CometExecIterator.executeNative(CometExecIterator.scala:65)
+at org.apache.comet.CometExecIterator.getNextBatch(CometExecIterator.scala:111)
+at org.apache.comet.CometExecIterator.hasNext(CometExecIterator.scala:126)
+
+...
+```
+Note:
+- The backtrace coverage in Datafusion is still improving. So there is a chance the error still not covered, feel free to file a [ticket](https://github.com/apache/arrow-datafusion/issues)
```
Comment (viirya) [inline DEBUGGING.md:154]:
```suggestion
- The backtrace coverage in DataFusion is still improving. So there is a chance the error still not covered, feel free to file a [ticket](https://github.com/apache/arrow-datafusion/issues)
```

File: DEBUGGING.md [hunk 5]
```diff
@@ -94,3 +94,62 @@ https://mail.openjdk.org/pipermail/hotspot-dev/2019-September/039429.html
...
+- The backtrace doesn't come for free and therefore intended for debugging purposes
```
Comment (viirya) [inline DEBUGGING.md:155]:
```suggestion
- The backtrace doesn't come for free and therefore intended for debugging purposes.
```

Comment (viirya) [reply DEBUGGING.md:155]: I'm afraid that "doesn't come for free" reads a bit ambiguous. Maybe change to more explicit way like "The backtrace comes with performance cost ...".

**Review by viirya — APPROVED** [review]

**Comment by comphead** [comment]
Addressed all the comments, thanks folks for helping with the review. I plan to merge it within 1 hr, if no other comments

**Pull request merged**